### PR TITLE
MOD-543: Unique POH challenge updates

### PR DESCRIPTION
**Pull request by rgovindji**
---
### Summary

This change introduces a unique poh challenge type and adds additional POH review status "processing" which is used for when the video is being processed. 

### Issues
*Link to the issue

### Type of Change
- [ ] Bug fix
- [x ] New feature
- [ ] Breaking change
- [ ] Documentation update

### Checklist
- [ x] I have performed a self-review of my own code.
- [ x] I have commented my code, particularly in hard-to-understand areas.
- [ x] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published.

### Test
*How did you test your change?

### Additional Context
*Add any other context or screenshots about the pull request

### CODE REVIEW

File: scripts/deployment/deployment_utils.sh [hunk 1]
```diff
@@ -144,11 +175,13 @@ function deploy_wallet_canister() {
           controller_id = principal "'${ARCHIVE_CONTROLLER}'";
         }
   }})'
-  wallet_arg="'$wallet_arg'"
+  #wallet_arg="'$wallet_arg'"
```
Comment (xwdeng) [inline scripts/deployment/deployment_utils.sh:178]:
```suggestion
```

File: scripts/infra/up_infra.sh [hunk 2]
```diff
@@ -43,6 +43,7 @@ function create_qa_canisters() {
   dfx canister create vesting_qa &&
   dfx canister create modclub_qa_assets &&
   dfx canister create airdrop_qa &&
+  dfx canister create decideid_qa_assets &&
```
Comment (xwdeng) [inline scripts/infra/up_infra.sh:46]: Should we rely on decideid repo to create this caniter ?

File: src/modclub/main.mo [hunk 3]
```diff
@@ -1533,86 +1534,170 @@ shared ({ caller = deployer }) actor class ModClub(env : CommonTypes.ENV) = this
...
+        throw Error.reject("POH Lambda key is not provided. Please ask admin to set the POH_LAMBDA_KEY for lambda calls.");
+      };
+
+      // Initiate lambda trigger to process face
+      try {
+        await pohEngine.httpCallForProcessing(
+          caller,
+          dataCanisterId!,
+          contentId!,
+          "dev", // TODO: Dynamically set this
```
Comment (VitalyGitDev) [inline src/modclub/main.mo:1680]: its good to have task in Linear for this

File: src/modclub/service/poh/poh.mo [hunk 4]
```diff
@@ -674,7 +687,8 @@ module PohModule {
                 return #alreadySubmitted;
               } else if (attempts.get(attempts.size() -1).status == #verified) {
                 return #alreadyApproved;
-              } else if (attempts.get(attempts.size() -1).status == #rejected) {
+              } else if (attempts.get(attempts.size() -1).status == #rejected or 
+              attempts.get(attempts.size() -1).status == #rejectedDuplicate) {
```
Comment (VitalyGitDev) [inline src/modclub/service/poh/poh.mo:691]: attempts is a buffer so expression like 
`attempts.get(attempts.size() -1)`
will crush the app if `buffer.size() // => 0`

File: src/modclub_assets/src/components/app/ModclubApp.tsx [hunk 5]
```diff
@@ -214,12 +213,20 @@ export default function ModclubApp() {
...
     ) : (
       <NotAuthenticatedModal />
     );
+  }
+
+  if (appState.userProfile && status && status !== "verified" && !selectedProvider && token) {
+    console.log("Debug Info - UserProfile:", appState.userProfile);
+    console.log("Debug Info - Status:", status);
+    console.log("Debug Info - Selected Provider:", selectedProvider);
+    console.log("Debug Info - Token:", token);
```
Comment (VitalyGitDev) [inline src/modclub_assets/src/components/app/ModclubApp.tsx:228]: Not sure about this debug line

File: src/modclub_assets/src/components/app/modals/UserIncompleteModal.tsx [hunk 6]
```diff
@@ -54,24 +54,18 @@ export default function UserIncompleteModal({
...
+            <>
+              <p>
+                Your Proof of Humanity has been rejected, duplicate account detected.
+              </p>
             </>
           )}
         </Modal.Card.Body>
         <Modal.Card.Footer className="pt-0" justifyContent="flex-end">
-          {status !== "pending" && status !== "verified" && (
+          {!(status === "pending" || status === "processing" || status === "verified" || status === "rejectedDuplicate") && (
```
Comment (VitalyGitDev) [inline src/modclub_assets/src/components/app/modals/UserIncompleteModal.tsx:68]: Its good to have Statuses as a constants, so maybe it worth to create a task for such refactoring

File: src/modclub_assets/src/utils/util.ts [hunk 7]
```diff
@@ -210,6 +210,9 @@ export const formattedTime = (val) => {
 };
 
 export const hideStringWithStars = (str) => {
+  if (str == null || str === undefined) {
```
Comment (VitalyGitDev) [inline src/modclub_assets/src/utils/util.ts:213]: I beleave its possible to simplify to 
```
if (!str || !str.length) {
    ....
}
```

**Review by VitalyGitDev — APPROVED** [review]

**Pull request merged**